File: time-management-advisor/src/main/java/com/sbnz/timemanagementadvisor/model/ActivityInstance.java
package com.sbnz.timemanagementadvisor.model;

import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
import com.sbnz.timemanagementadvisor.model.enums.ActivityType;
import lombok.Data;
import org.bson.types.ObjectId;

import java.time.LocalTime;
import java.util.List;

@Data
public class ActivityInstance {
    @JsonSerialize(using = ToStringSerializer.class)
    private ObjectId activityId;
    private String title, description;
    private List<CheckListItem> checklist;
    private LocalTime start, end;
    private ActivityType activityType;
    private List<LocalTime> monitoring;
}

File: frontend/time-management-app/src/model/ActivityInstance.ts
import { CheckListItem } from "./CheckListItem";
import { ActivityType } from "./enums/ActivityType";

export interface ActivityInstance {
  activityId: string;
  title: string;
  description: string;
  checklist: CheckListItem[];
  start: string; //Date;
  end: string; //Date;
  activityType: ActivityType;
}
